src/plugins/intel_gpu/src/plugin/transformations/broadcast_reshape_matmul_fusion.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "broadcast_reshape_matmul_fusion.hpp"
+
+#include "intel_gpu/op/gemm.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+BroadcastReshapeMatmulFusion::BroadcastReshapeMatmulFusion() {
+    using namespace ov::pass::pattern;
+
+    auto not_reshape = [](const ov::Output<ov::Node>& output) -> bool {
+        return std::dynamic_pointer_cast<ov::op::v1::Reshape>(output.get_node_shared_ptr()) == nullptr;
+    };
+
+    auto broadcast_rank_equals_and_has_static_dims = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(5)(output) && has_static_dims({2, 3}) && consumers_count(1);
+    };
+
+    auto reshape_rank_equals_and_has_static_dim = [](const ov::Output<ov::Node>& output) -> bool {
+        return rank_equals(4)(output) && has_static_dim(2) && consumers_count(1);
+    };
+
+    auto input_a_m = any_input(not_reshape);
+    auto input_b_m = any_input(not_reshape);
+
+    auto broadcast_a_target_shape_m = wrap_type<ov::op::v0::Constant>();
+    auto broadcast_a_m = wrap_type<ov::op::v3::Broadcast>({input_a_m, broadcast_a_target_shape_m}, broadcast_rank_equals_and_has_static_dims);
```

Comment:
The pattern should probably be extended. Usually it looks like this:
`Concat(KVCache) -> Unsqueeze/Reshape -> Broadcast -> Squeeze/Reshape -> MatMul/Gemm`
So you should also capture first Unsqueeze/Reshape and remove it. KVCache probably is not needed, but can also be captured if we want to limit applicability to GQA only

Reply:
The expansion to pattern that includes pre `unsqueeze` will also be carried out separately by including it in the item list in the next task.